src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceTagsTest.java

```diff
@@ -0,0 +1,443 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.github.tomakehurst.wiremock.common.FileSource;
+import com.github.tomakehurst.wiremock.common.SingleRootFileSource;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.extension.ResponseTransformer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.Response;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.github.tomakehurst.wiremock.stubbing.Scenario;
+import jenkins.scm.api.SCMHeadObserver;
+import jenkins.scm.api.mixin.ChangeRequestCheckoutStrategy;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.kohsuke.github.GHFileNotFoundException;
+import org.kohsuke.github.GHException;
+import org.kohsuke.github.GHRef;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.PagedIterator;
+import org.kohsuke.github.PagedIterable;
+import org.mockito.Mockito;
+
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class GithubSCMSourceTagsTest {
+
+    /**
+     * All tests in this class only use Jenkins for the extensions
+     */
+    @ClassRule
+    public static JenkinsRule r = new JenkinsRule();
+
+    public static WireMockRuleFactory factory = new WireMockRuleFactory();
+
+    @Rule
+    public WireMockRule githubRaw = factory.getRule(WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderClasspath("raw")
+    );
+    @Rule
+    public WireMockRule githubApi = factory.getRule(WireMockConfiguration.options()
```

Comment:
Now that you have more than one test class using this, pull this out into a base class.

Reply:
But this can also wait to a new PR.